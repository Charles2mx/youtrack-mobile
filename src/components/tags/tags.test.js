import React from 'react';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import Tags from './tags';

describe('<Tags/>', () => {

  let wrapper;

  describe('Render', () => {
    beforeEach(() => {
      wrapper = doShallow([{color: {id: 1}}]);
    });

    it('should match a snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component', () => {
      expect(findByTestId('tags')).toHaveLength(1);
    });

    it('should render color field', () => {
      expect(findByTestId('tagColor')).toHaveLength(1);
    });
  });


  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(tags = [], onTagPress = () => {}) {
    return shallow(
      <Tags
        tags={tags}
        onTagPress={onTagPress}
      />
    );
  }
});
