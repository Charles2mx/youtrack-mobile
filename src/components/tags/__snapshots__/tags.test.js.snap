// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags/> Render should match a snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "marginLeft": -2,
        "marginRight": -2,
      },
      Object {},
    ]
  }
  testID="tags"
>
  <TouchableOpacity
    activeOpacity={0.2}
    key="undefined_button"
    onPress={[Function]}
    testID="tagsTag"
  >
    <ColorField
      color={
        Object {
          "id": 1,
        }
      }
      fullText={true}
      style={
        Array [
          Object {
            "margin": 2,
            "paddingLeft": 4,
            "paddingRight": 4,
            "width": null,
          },
          Object {},
        ]
      }
      testID="tagColor"
    />
  </TouchableOpacity>
</View>
`;
